log-manager/src/test/java/io/airlift/log/TestJsonFormatter.java

```diff
@@ -0,0 +1,70 @@
+package io.airlift.log;
+
+import io.airlift.json.JsonCodec;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.logging.LogRecord;
+
+import static com.google.common.base.Throwables.getStackTraceAsString;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.time.temporal.ChronoUnit.NANOS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestJsonFormatter
+{
+    @Test
+    public void testMinimalJsonErrorLogLine()
+    {
+        JsonRecord originalJsonRecord = new JsonRecord(Instant.now(), Level.DEBUG, "thread-0", "TestLogger", "Test Log Message", new Exception("Test Exception 1"));
+
+        RuntimeException exception = new RuntimeException("Test Exception 2");
+        String minimalJsonErrorLogLine = (new JsonFormatter()).minimalJsonErrorLogLine(originalJsonRecord, exception);
+
+        assertTrue(minimalJsonErrorLogLine.endsWith("\n"), "Log lines should end with newline");
+
+        assertEquals(
+                jsonCodec(JsonRecord.class).fromJson(minimalJsonErrorLogLine),
+                new JsonRecord(originalJsonRecord.getTimestamp(), Level.ERROR, null, null, exception.getMessage(), null));
+    }
+
+    @Test
+    public void testRoundTrip()
+    {
+        JsonRecord originalJsonRecord = new JsonRecord(Instant.now(), Level.DEBUG, "thread-0", "TestLogger", "Test Log Message", new Exception("Test Exception 1"));
+        JsonCodec<JsonRecord> jsonRecordJsonCodec = jsonCodec(JsonRecord.class);
+
+        assertEquals(
+                jsonRecordJsonCodec.fromJson(jsonRecordJsonCodec.toJson(originalJsonRecord)),
+                new JsonRecord(originalJsonRecord.getTimestamp(), originalJsonRecord.getLevel(), originalJsonRecord.getThread(), originalJsonRecord.getLoggerName(), originalJsonRecord.getMessage(), null));
+    }
+
+    @Test
+    public void testLogFormatting()
+    {
+        Exception testException = new RuntimeException("Test Exception");
+        LogRecord record = new LogRecord(Level.DEBUG.toJulLevel(), "Test Log Message");
+        record.setLoggerName("TestLogger");
+        record.setThrown(testException);
+
+        String logMessage = (new JsonFormatter()).format(record);
+
+        Map<String, String> jsonMap = jsonCodec(Map.class).fromJson(logMessage);
```

Comment:
I'm surprised this compiles, since `Map` is a raw type, but it does with an unchecked assignment warning. You can make this type safe with
```java
mapJsonCodec(String.class, String.class)